Gems/Atom/Tools/MaterialEditor/Code/Source/Document/MaterialDocument.cpp

```diff
@@ -262,181 +191,78 @@ namespace MaterialEditor
             property.SetConfig(propertyConfig);
         }
 
-        // Auto add or checkout saved file
-        AzToolsFramework::SourceControlCommandBus::Broadcast(&AzToolsFramework::SourceControlCommandBus::Events::RequestEdit,
-            m_absolutePath.c_str(), true, [](bool, const AzToolsFramework::SourceControlFileInfo&) {});
-
-        AZ_TracePrintf("MaterialDocument", "Material document saved: '%s'.\n", m_absolutePath.data());
-
-        AtomToolsFramework::AtomToolsDocumentNotificationBus::Broadcast(&AtomToolsFramework::AtomToolsDocumentNotificationBus::Events::OnDocumentSaved, m_id);
-
-        m_saveTriggeredInternally = true;
-        return true;
+        return SaveSucceeded();
     }
 
     bool MaterialDocument::SaveAsCopy(AZStd::string_view savePath)
     {
-        using namespace AZ;
-        using namespace RPI;
-
-        if (!IsOpen())
-        {
-            AZ_Error("MaterialDocument", false, "Material document is not open to be saved: '%s'.", m_absolutePath.c_str());
-            return false;
-        }
-
-        if (!IsSavable())
-        {
-            AZ_Error("MaterialDocument", false, "Material types can only be saved as a child: '%s'.", m_absolutePath.c_str());
-            return false;
-        }
-
-        AZStd::string normalizedSavePath = savePath;
-        if (!AzFramework::StringFunc::Path::Normalize(normalizedSavePath))
+        if (!AtomToolsDocument::SaveAsCopy(savePath))
         {
-            AZ_Error("MaterialDocument", false, "Material document save path could not be normalized: '%s'.", normalizedSavePath.c_str());
             return false;
```

Comment:
[question] Why does this `return false;` instead of `return SaveFailed();` like the other returns in this function?

Same in other locations where the first return is false but the rest is `return SOMETHINGFailed();`. SOMETHING being save, open, etc.

Reply:
I'm going to do another pass to update all of be open and save functions to return an enum result instead of just true or false. It should have states for succeeded, failed, in progress/incomplete. If I'm returning true or false without calling succeeded or failed functions, it's just to avoid triggering notifications or printing messages multiple times.